lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts

```diff
@@ -0,0 +1,614 @@
+    (
+      time: JulianDate | undefined
+    ): ArcGisImageServerImageryProvider | undefined => {
+      if (!isDefined(this.url)) {
+        return undefined;
+      }
+
+      const params = { ...this.flattenedParameters };
+      if (time) params.time = JulianDate.toDate(time).getTime();
+
+      let tilingScheme: WebMercatorTilingScheme | GeographicTilingScheme;
+
+      if (this.wkid === 102100 || this.wkid === 102113 || this.wkid === 3857) {
+        tilingScheme = new WebMercatorTilingScheme();
+      } else if (this.wkid === 4326) {
+        tilingScheme = new GeographicTilingScheme();
+      } else {
+        throw TerriaError.from(
+          `Tile spatial reference WKID ${this.wkid} is not supported.`
+        );
+      }
+
+      return new ArcGisImageServerImageryProvider({
+        url: cleanAndProxyUrl(this, this.url),
+        tilingScheme: tilingScheme,
+        maximumLevel: this.maximumLevel,
+        minimumLevel: this.minimumLevel,
+        tileHeight: this.tileHeight,
+        tileWidth: this.tileWidth,
+        parameters: params,
+        enablePickFeatures: this.allowFeaturePicking,
+        usePreCachedTiles: this.usePreCachedTiles,
+        token: this.token,
+        credit: this.attribution ?? ""
+      });
+    }
+  );
+
+  @computed
+  get mapItems() {
+    if (this.isLoadingMetadata) return [];
+    return [this._currentImageryParts, this._nextImageryParts].filter(
+      isDefined
+    );
+  }
+
+  @override
+  get selectableDimensions() {
+    return filterOutUndefined([
+      ...super.selectableDimensions,
+      this.rasterFunctionSelectableDimensions
+    ]);
+  }
+
+  @computed
+  get rasterFunctionSelectableDimensions():
+    | SelectableDimensionEnum
+    | undefined {
+    if (this.disableRasterFunctionSelectors) return undefined;
+    return {
+      id: "raster-functions",
+      name: i18next.t("models.arcGisImageServerCatalogItem.rasterFunction"),
+      options: this.availableRasterFunctions.map((rasterFn) => ({
+        id: rasterFn.name,
+        name: rasterFn.name,
+        description: rasterFn.description
+      })),
+      selectedId: this.renderingRule?.rasterFunction,
+      allowUndefined: true,
+      undefinedLabel: "Default",
+      setDimensionValue: (strata, rasterFunction) => {
+        if (!this.renderingRule) {
+          this.setTrait(
+            strata,
+            "renderingRule",
+            createStratumInstance(ArcGisImageServerRenderingRule, {
+              rasterFunction
+            })
+          );
+        } else {
+          this.renderingRule.setTrait(strata, "rasterFunction", rasterFunction);
+        }
+      }
+    };
+  }
+}
+
+function getBaseURI(item: ArcGisImageServerCatalogItem) {
+  const uri = new URI(item.url);
+  const lastSegment = uri.segment(-1);
+  if (lastSegment && lastSegment.match(/\d+/)) {
+    uri.segment(-1, "");
+  }
+  return uri;
+}
+
+async function getJson(item: ArcGisImageServerCatalogItem, uri: any) {
```

Comment:
Nitpick: Instead of `any` we could use `uri.URI` type.

Reply:
Done!